clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1415,6 +1415,157 @@ def VTableAddrPointOp : CIR_Op<"vtable.address_point",
   let hasVerifier = 1;
 }
 
+
+//===----------------------------------------------------------------------===//
+// SetBitfieldOp
+//===----------------------------------------------------------------------===//
+
+def SetBitfieldOp : CIR_Op<"set_bitfield"> {
+  let summary = "Set a bitfield";
+  let description = [{ 
+    The `cir.set_bitfield` operation provides a store-like access to 
+    a bit field of a record.
+
+    It expects an address of a storage where to store, a value being stored,
+    a type of a bitfield, a name of a bit field, a pointer to a storage in the
+    base record, a size of the storage, a size the bit field, an offset 
+    of the bit field and a sign. Returns a value being stored.
+
+    Example:
+    ```mlir
+    // Suppose we have a struct with multiple bitfields stored in
+    // different storages. The `cir.get_bitfield` operation gets the value
+    // of the bitfield
+
+    !u32i = !cir.int<u, 32>
+    !u64i = !cir.int<u, 64>
+    !ty_22A22 = !cir.struct<struct "A" {!u64i, !u32i}>
```

Comment:
The struct doesn't seem to be used, for the example sake, you could use  `!structA` or something less mangled, probably easier to read. Perhaps also add the operation that's using this type as part of the example?